### PR TITLE
Run commands with env vars defined in the bundle config

### PR DESCRIPTION
**Pull request by vanstee**
This allows bundle authors to include environment variables for each command in their bundle configs which are included in the environment when running the command. They are overridden by any dynamic config if available as it is applied afterwards. No change in Cog is necessary since we have already been expecting it in our bundle config schema.

### CODE REVIEW

**Comment by vanstee** [comment]
Looks like some `go fmt` spacing snuck in there. Sorry about that.

File: relay/messages/execution.go [hunk 1]
```diff
@@ -17,7 +17,6 @@ type ExecutionRequest struct {
 	InvocationID   string                 `json:"invocation_id"`
 	InvocationStep string                 `json:"invocation_step"`
 	Command        string                 `json:"command"`
-	CommandConfig  map[string]interface{} `json:"command_config"`
```
Comment (christophermaier) [inline relay/messages/execution.go:20]: Why was this removed? Was this dead code?

File: relay/messages/execution.go [hunk 2]
```diff
@@ -73,11 +72,11 @@ var errorCommandNotFound = errors.New("Command not found")
 // ToCircuitRequest converts an ExecutionRequest into a circuit.api.ExecRequest
 func (er *ExecutionRequest) ToCircuitRequest(bundle *config.Bundle, relayConfig *config.Config, useDynamicConfig bool) (*api.ExecRequest, bool, error) {
 	retval := &api.ExecRequest{}
-	hasDynamicConfig := er.compileEnvironment(retval, relayConfig, useDynamicConfig)
-	command := bundle.Commands[er.commandName]
+	command := bundle.Commands[er.CommandName()]
 	if command == nil {
 		return nil, false, errorCommandNotFound
 	}
+	hasDynamicConfig := er.compileEnvironment(command, retval, relayConfig, useDynamicConfig)
```
Comment (christophermaier) [inline relay/messages/execution.go:79]: Can we rename this variable? It's a little misleading now, seeing as how we're doing more than just dealing with dynamic config. Maybe something like `withEnvironment`, or similar?

File: relay/messages/execution.go [hunk 3]
```diff
@@ -73,11 +72,11 @@ var errorCommandNotFound = errors.New("Command not found")
 // ToCircuitRequest converts an ExecutionRequest into a circuit.api.ExecRequest
 func (er *ExecutionRequest) ToCircuitRequest(bundle *config.Bundle, relayConfig *config.Config, useDynamicConfig bool) (*api.ExecRequest, bool, error) {
 	retval := &api.ExecRequest{}
-	hasDynamicConfig := er.compileEnvironment(retval, relayConfig, useDynamicConfig)
-	command := bundle.Commands[er.commandName]
+	command := bundle.Commands[er.CommandName()]
```
Comment (christophermaier) [inline relay/messages/execution.go:75]: How did this work before?

File: relay/messages/execution.go (hunk 1 above)
Comment (vanstee) [reply relay/messages/execution.go:20]: This was an unused config var yes. I'm guessing it was meant to include env vars for commands, but we already have those in the config anyway so we don't need it with the execution request.

File: relay/messages/execution.go (hunk 3 above)
Comment (vanstee) [reply relay/messages/execution.go:75]: I reordered the call to `compileEnvironment`. Before after `compileEnvironment` was called that attribute would be set. Now I'm just pulling out the command first, moving up the error, and then sending that in to `compileEnvironment` to include extra env vars if needed.

File: relay/messages/execution.go (hunk 2 above)
Comment (vanstee) [reply relay/messages/execution.go:79]: That's still the same `hasDynamicConfig` boolean. The env actually gets set on `retval`, which seems plain but still properly named.

**Comment by vanstee** [comment]
@christophermaier Let me know if that answers everything.

File: relay/messages/execution.go (hunk 2 above)
Comment (christophermaier) [reply relay/messages/execution.go:79]: ah, ok

**Review by christophermaier — APPROVED** [review]

**Comment by christophermaier** [comment]
Fixes operable/cog#1143

**Comment by agis** [comment]
Aww 🎉 ❤️
